electron/execution.ts

```diff
@@ -90,51 +94,58 @@ let browserContext: BrowserContext | null = null;
 let actionListener = new EventEmitter();
 let isBrowserRunning = false;
 
-async function onRecordJourneys(data: { url: string }, browserWindow: BrowserWindow) {
-  try {
+function onRecordJourneys(mainWindowEmitter: EventEmitter) {
+  return async function (data: { url: string }, browserWindow: BrowserWindow) {
     if (isBrowserRunning) {
       throw new Error(
         'Cannot start recording a journey, a browser operation is already in progress.'
       );
     }
     isBrowserRunning = true;
-    const { browser, context } = await launchContext();
-    browserContext = context;
-    actionListener = new EventEmitter();
-    // Listen to actions from Playwright recording session
-    const actionsHandler = (actions: ActionInContext[]) => {
-      ipc.callRenderer(browserWindow, 'change', { actions });
-    };
-    actionListener.on('actions', actionsHandler);
-
-    // _enableRecorder is private method, not defined in BrowserContext type
-    await (context as any)._enableRecorder({
-      launchOptions: {},
-      contextOptions: {},
-      startRecording: true,
-      showRecorder: false,
-      actionListener,
-    });
-    await openPage(context, data.url);
-
-    const closeBrowser = async () => {
-      browserContext = null;
-      actionListener.removeListener('actions', actionsHandler);
-      try {
-        await browser.close();
-      } catch (e) {
-        logger.error('Browser close threw an error', e);
-      }
-    };
+    try {
+      const { browser, context } = await launchContext();
+      const closeBrowser = async () => {
+        browserContext = null;
+        actionListener.removeListener('actions', actionsHandler);
+        try {
+          await browser.close();
+        } catch (e) {
+          logger.error('Browser close threw an error', e);
+        }
+      };
+      ipc.addListener('stop', closeBrowser);
+      // Listen to actions from Playwright recording session
+      const actionsHandler = (actions: ActionInContext[]) => {
+        ipc.callRenderer(browserWindow, 'change', { actions });
+      };
+      browserContext = context;
+      actionListener = new EventEmitter();
+      actionListener.on('actions', actionsHandler);
 
-    ipc.on('stop', closeBrowser);
+      mainWindowEmitter.addListener(MainWindowEvent.MAIN_CLOSE, async () => {
+        actionListener.removeAllListeners();
+        ipc.removeListener('stop', closeBrowser);
+        return browser.close().then(() => {
+          isBrowserRunning = false;
```

Comment:
Not so important comment - as the function is async, shall we use await here?
```suggestion
       await browser.close();
       isBrowserRunning = false;
```

Reply:
0278c46d9e0c345560c521c8707907e6d62c94d4